packages/compass-indexes/src/modules/create-index.tsx

```diff
@@ -337,6 +359,106 @@ export type CreateIndexSpec = {
   [key: string]: string | number;
 };
 
+type SuggestedIndexesRequestedAction = {
+  type: ActionTypes.SuggestedIndexesRequested;
+};
+
+export type SuggestedIndexFetchedAction = {
+  type: ActionTypes.SuggestedIndexesFetched;
+  sampleDocs: Array<Document>;
+  indexSuggestions: { [key: string]: number } | null;
+  fetchingSuggestionsError: string | null;
+  indexSuggestionsState: IndexSuggestionState;
+};
+
+export type SuggestedIndexFetchedProps = {
+  dbName: string;
+  collectionName: string;
+  inputQuery: string;
+};
+
+export const fetchIndexSuggestions = ({
+  dbName,
+  collectionName,
+  inputQuery,
+}: {
+  dbName: string;
+  collectionName: string;
+  inputQuery: string;
+}): IndexesThunkAction<
+  Promise<void>,
+  SuggestedIndexFetchedAction | SuggestedIndexesRequestedAction
+> => {
+  return async (dispatch, getState, { dataService }) => {
+    dispatch({
+      type: ActionTypes.SuggestedIndexesRequested,
+    });
+    const namespace = `${dbName}.${collectionName}`;
+
+    // Get sample documents from state if it's already there, otherwise fetch it
+    let sampleDocuments: Array<Document> | null =
+      getState().createIndex.sampleDocs || null;
+
+    // If it's null, that means it has not been fetched before
+    if (sampleDocuments === null) {
+      try {
+        sampleDocuments =
+          (await dataService.sample(namespace, { size: 50 })) || [];
```

Comment:
I should mention, ideally we'd be passing an abort signal here, that way we can cancel the operation if the user closes the tab or something. I think this is fine as is for now, but something to keep in mind for future asynchronous things that allow for abort signals.